### PR TITLE
List literals

### PR DESCRIPTION
**Pull request by UlfNorell**
(on top of #35)

### CODE REVIEW

**Review by omelkonian — APPROVED** [review]
Much more idiomatic, indeed! 

Another way to go about it is support both, by providing appropriate syntax on the Agda side.
For example, you can do [this trick](https://github.com/omelkonian/formal-prelude/blob/master/Prelude/Nary.agda#L8-L13) and the users will be able to choose whatever they prefer.

**Comment by UlfNorell** [comment]
Only on the right-hand side though.

File: lib/Haskell/Prelude.agda [hunk 1]
```diff
@@ -116,6 +125,16 @@ case x of f = f x
 asTypeOf : a → a → a
 asTypeOf x _ = x
 
+undefined : {@(tactic absurd) i : ⊥} → a
+undefined {i = ()}
+
+error : {@(tactic absurd) i : ⊥} → String → a
```
Comment (jmchapman) [inline lib/Haskell/Prelude.agda:131]: this is really cool!

File: test/Test.agda [hunk 2]
```diff
@@ -159,13 +163,9 @@ sumMon []       = mempty
...
 {-# COMPILE AGDA2HS sumMon #-}
 
-data NonEmpty {a : Set} : List a → Set where
-  instance
-    itsNonEmpty : ∀ {x xs} → NonEmpty (x ∷ xs)
-
 -- Instance argument proof obligation that should not turn into a class constraint
 hd : (xs : List a) → ⦃ NonEmpty xs ⦄ → a
-hd [] ⦃ () ⦄
+hd []      = error "hd: empty list"
```
Comment (jmchapman) [inline test/Test.agda:168]: this is very cool - best of both worlds.

**Pull request merged**